### ISSUE TITLE
rename master branch to something less offensive

### ISSUE THREAD

**Issue by j0hnL** [helm/helm#8412]
Many software projects that are managed and distributed using tools like git use master/dev terminology. The term master has extremely offensive connotations, and the team at GitHub agrees (https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/). We should encourage all software projects to adopt less offensive terminology.

The primary software project I lead has switched to using the term release, instead of master. Since the master branch is mostly used for offering software releases, this seemed to make the most sense. Another term that could replace master is main.

**Comment by bacongobbler** [comment]
I would not be opposed to switching the branch name to `main` or `trunk`.

There would be documentation to update, of course. But I don't think there's any immediate risk changing the branch name.

**Comment by bacongobbler** [comment]
We discussed this during the dev call. Those in attendance were all in agreement this would be a good idea to pursue, however there were concerns regarding renaming the branch and whether all of the PRs open at https://github.com/helm/helm/pulls would have to be re-targeted or re-opened.

Another point raised was whether we should document the process. Other projects in the CNCF are likely having similar conversations, or will be at some point in the future.

The call to action at the end of the call was to do some more research on what renaming `master` to `main` may entail in terms of work items. That way we have a clearer picture of what might be involved here and we can coordinate efforts.

@j0hnL have you seen other large projects go through with this work, and have they documented their process somewhere?

**Comment by j0hnL** [comment]
@bacongobbler we edited the base branch of outstanding PR:  https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request

**Comment by hickeyma** [comment]
Thanks for the link @j0hnL. Thanks to feedback from @paulczar, it would seem that GH are currently working to provide help/tooling etc. to support renaming. In light of this, I wonder if it would be worth waiting for this support first before progressing?

**Comment by mattfarina** [comment]
First, I am in favor of renaming.

What we have to deal with, I think, is the logistics of carrying it out. I ask that we have a little patience in doing this because:

- GitHub is working on this problem as a company. If they have some automation or direction that's readily picked up by people it would be a good experience if we (Helm maintainers) could use it and contributors could have a familiar naming experience.
- Kubernetes is doing a Naming working group. I imagine other sister CNCF projects are also in the same position. It would be good if we could do some coordination to ease the experience on people. The same name would make things easier.
- Linux just brought in a policy on names. Linus is the creator of Git along with Linux. I would like some time to see what their policy says and means. Linux is another Linux Foundation project like the rest of the CNCF ones.

Basically, I think it would be great if we did this in a thoughtful way and that will require a little patience.

**Comment by mattfarina** [comment]
FYI, I've reached out to the CNCF on this. They are in the best position to learn what's happening with GH, know about other LF projects, and to coordinate across CNCF projects.

**Comment by hickeyma** [comment]
Thanks for the update @mattfarina. It is good to get some background. 

I am also in favour of renaming.

**Comment by hickeyma** [comment]
FYI, GH will support the transition from master to main. For more details: https://github.com/github/renaming.

Thanks @sdake for the heads up.

**Comment by mattfarina** [comment]
I would suggest we hold off until the "seamless move for existing repositories" capabilities. There are many moving parts and repos for Helm. This would be the simplest path.

**Comment by mattfarina** [comment]
@hickeyma thanks for sharing that renaming link. I was looking something about it the other day and hadn't realized it was there.

**Comment by bacongobbler** [comment]
It looks like this feature is now available. We just need to find a time that works for a majority of maintainers to ensure no outages occur after the rename.

https://github.com/github/renaming#renaming-existing-branches

> Renaming a branch will:

> - Re-target any open pull requests
> - Update any draft releases based on the branch
> - Move any branch protection rules that explicitly reference the old name
> - Update the branch used to build GitHub Pages, if applicable
> - Show a notice to repository contributors, maintainers, and admins on the repository homepage with instructions to update local copies of the repository
> - Show a notice to contributors who git push to the old branch
> - Redirect web requests for the old branch name to the new branch name
> - Return a "Moved Permanently" response in API requests for the old branch name

**Comment by hickeyma** [comment]
Default branch is now renamed from `master` to `main`. 

"GitHub will automatically redirect links on GitHub.com that contain the old branch name to the equivalent link on the renamed branch. GitHub will also update branch protection policies, as well as the base branch for open pull requests and draft releases." Refer to [Renaming a branch](https://docs.github.com/en/github/administering-a-repository/renaming-a-branch) for more details.

If you have a local clone of the repository you will need to update the clone. From your local clone of the repository, run the following commands to update the name of the default branch:

```
git branch -m master main
git fetch origin
git branch -u origin/main main
git remote set-head origin -a
```

Thanks @j0hnL for starting the discussion with the community.

**Comment by hickeyma** [comment]
Closing as branch is renamed and PR #9549 updates any references to the default branch name.

**Comment by ph0uad** [comment]
Is there an example of someone expressing genuine discontent over this?
